### PR TITLE
feat(ADM-1379): Migrate harvest-client From CircleCi To GHA

### PR DESCRIPTION
**Pull request by PredragMPetrovic**
### Checklist

- [x] I've run `./gradlew test` from the root directory to see all new and existing tests pass
- [x] I've followed the _harvest-client_ code style
- [x] I've read the [Contribution Guidelines][contribution guidelines]
- [x] I've updated the documentation if necessary.

### Motivation and Context

Motivation is to move away from CirlcleCI to github actions, since we don't want to manage project on CircleCI anymore and github actions are a lot faster, with fewer lines of code and a lot of stuff provided out of the box.

### Description

In this PR two scripts are provided:
- pull-request.yml -> in this one we build project, run all the tests and submit results to codecov.io. It happens on opening PR on develop branch
- build-and-publish-image.yml -> here we are building project and image and published it (signed) to maven, this happens on merging branch to develop branch

[contribution guidelines]: ../CONTRIBUTING.md

### CODE REVIEW

File: .github/workflows/build-and-publish-image.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Publish Docker image
```
Comment (mnembrini) [inline .github/workflows/build-and-publish-image.yml:1]: We are publishing a Jar / to maven central not a docker image

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  pull_request:
+    branches:
+      - develop
+
+jobs:
+  gradle:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Cheeckout
```
Comment (mnembrini) [inline .github/workflows/pull-request.yml:13]:
```suggestion
    - name: Checkout
```

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        HARVEST_AUTH_TOKEN_ADMIN1: ${{ secrets.HARVEST_AUTH_TOKEN_ADMIN1 }}
+        HARVEST_ACCOUNT_ID_ADMIN2: ${{ secrets.HARVEST_ACCOUNT_ID_ADMIN2 }}
+        HARVEST_AUTH_TOKEN_ADMIN2: ${{ secrets.HARVEST_AUTH_TOKEN_ADMIN2 }}
+      with:
+        gradle-version: 7.2
+        arguments: build
+
+    - name: Generate test coverage report
+      run: |-
+        echo "Generating test coverage report"
```
Comment (mnembrini) [inline .github/workflows/pull-request.yml:36]: use gradle-build-action

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        ./gradlew jacocoTestReport
+
+    - name: Upload report to codecov.io
+      run: |-
+        echo "Uploading report..."
+        bash <(curl -s https://codecov.io/bash) -f build/reports/jacoco/test/jacocoTestReport.xml
+
+    - name: Save test results
```
Comment (mnembrini) [inline .github/workflows/pull-request.yml:44]: Does this work on GHA?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,5 @@
 # A Harvest Java Client
 
-[![CircleCI](https://circleci.com/gh/3AP-AG/harvest-client.svg?style=svg)](https://circleci.com/gh/3AP-AG/harvest-client)
```
Comment (mnembrini) [inline README.md:3]: Let's have a GHA badge here https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge of the latest `develop` branch

**Review by mnembrini — CHANGES_REQUESTED** [review]
Looks pretty good overall, some minor details to fix first

File: README.md [hunk 6]
```diff
@@ -119,21 +118,15 @@ dependencies {
...
+- Go to "Settings" -> "Chose Modules" -> Tick all modules
+- ! Only for the second account: Go to "Settings" -> "Edit Preferences" -> Set Time Mode to "Track time via start and end time"
+- Top left on your name -> My Profile -> link under work email to your Harvest ID settings -> Developers -> create new personal Token (name does not matter)
 Account ID and token are needed for authentication
-
-Add your token to the local files admin1.conf and admin2.conf under src/test/resources/admin*.conf
-
-Update the CircleCI project environment variables with these Id and tokens here:
-https://app.circleci.com/settings/project/github/3AP-AG/harvest-client/environment-variables
-
```
Comment (mnembrini) [inline README.md:136]: one more section of the readme is outdated: the publishing to Github packages. Please update it with the current status (we only publish master to maven central)

File: .github/workflows/pull-request.yml (hunk 4 above)
Comment (PredragMPetrovic) [reply .github/workflows/pull-request.yml]: Unfortunately not, I didn't manage to make it work with GHA, so I deleted this part.

**Review by mnembrini — APPROVED** [review]
Looks very nice now, thanks for the effort!

File: README.md (hunk 5 above)
Comment (primaproxima) [reply README.md:3]: Moved the `pull-request` badge to the first place.

File: .github/workflows/pull-request.yml (hunk 4 above)
Comment (primaproxima) [reply .github/workflows/pull-request.yml]: Are we talking about upload of artifacts, in this case the test results?
Yes, we this works on gha. Please take a look at [upload artifacts action](https://github.com/actions/upload-artifact).

**Review by primaproxima — APPROVED** [review]
Looks good. Just check the comment on uploading.

File: .github/workflows/pull-request.yml (hunk 4 above)
Comment (PredragMPetrovic) [reply .github/workflows/pull-request.yml]: We wanted something like this:

https://github.com/marketplace/actions/publish-unit-test-results

But I couldn't make it work.
It was reporting 'No tests' each time.

**Review by mmilic90 — APPROVED** [review]
Looks nice :)

**Pull request merged**